code/game/objects/effects/wanted_poster.dm

```diff
@@ -0,0 +1,56 @@
+/obj/item/weapon/poster/legit/wanted
+	var/poster_desc
+	icon_state = "rolled_poster"
+
+/obj/item/weapon/poster/legit/wanted/New(turf/loc, icon/person_icon, wanted_name, description)
+	..(loc)
+	name = "wanted poster ([wanted_name])"
+	desc = "A wanted poster for [wanted_name]."
+	poster_desc = description
+	qdel(resulting_poster)
+	if(person_icon)
+		var/obj/item/weapon/photo/photo = new()
+		photo.photocreate(null, icon(person_icon, dir = SOUTH))//copy this to prevent the cropping from happening to all the other images. There's probably a better way to copy an icon, but this works, so whatever.
+		resulting_poster = new /obj/structure/sign/poster/wanted(person_icon, wanted_name, poster_desc)
+		if(!resulting_poster || qdeleted(resulting_poster))
+			qdel(src)
+
+/obj/structure/sign/poster/wanted
+	var/wanted_name
+
+/obj/structure/sign/poster/wanted/New(var/icon/person_icon, var/person_name, var/description)
+	if(!person_icon)
+		qdel(src)
+		return
+	name = "wanted poster ([person_name])"
+	wanted_name = person_name
+	desc = description
+
+	var/obj/item/weapon/photo/photo = new /obj/item/weapon/photo()
+	photo.photocreate(null, icon(person_icon, dir = SOUTH))//copy the image so we don't mess with the one in the record.
+	person_icon = photo.img
+	qdel(photo)
+	var/icon/the_icon = icon("icon" = 'icons/obj/poster_wanted.dmi', "icon_state" = "wanted_background")
+	var/icon/icon_foreground = icon("icon" = 'icons/obj/poster_wanted.dmi', "icon_state" = "wanted_foreground")
+	person_icon.Shift(SOUTH, 7)
+	person_icon.Crop(7,4,26,30)
+	person_icon.Crop(-5,-2,26,29)
+	the_icon.Blend(person_icon, ICON_OVERLAY)
+	the_icon.Blend(icon_foreground, ICON_OVERLAY)
+
+	the_icon.Insert(the_icon, "wanted")
+	the_icon.Insert(icon('icons/obj/contraband.dmi', "poster_being_set"), "poster_being_set")
+	the_icon.Insert(icon('icons/obj/contraband.dmi', "poster_ripped"), "poster_ripped")
+	icon = the_icon
```

Comment:
Definitely a good thought, but I am not going to deal with that right now and have a chance of horribly breaking it.